lib/matplotlib/dviread.py

```diff
@@ -1000,21 +1003,173 @@ def _parse(self, file):
         return re.findall(br'/([^][{}<>\s]+)', data)
 
 
-def find_tex_file(filename, format=None):
+class _tex_support_cache(object):
+    """A persistent cache of data related to support files related to dvi
+    files produced by TeX. Currently holds results from :program:`kpsewhich`,
+    in future versions could hold pre-parsed font data etc.
+
+    Usage::
+
+      # create or get the singleton instance
+      cache = _tex_support_cache.get_cache()
+      cache.update_pathnames({"pdftex.map": "/usr/local/pdftex.map",
+                              "cmsy10.pfb": "/usr/local/fonts/cmsy10.pfb"})
+      pathnames = cache.get_pathnames(["pdftex.map", "cmr10.pfb"])
+      # now pathnames = {"pdftex.map": "/usr/local/pdftex.map"}
+
+      # optional after inserting new data, may improve query performance:
+      cache.optimize()
+
+    Parameters
+    ----------
+
+    filename : str, optional
+        File in which to store the cache. Defaults to `texsupport.db` in
+        the standard cache directory.
     """
-    Find a file in the texmf tree.
+
+    __slots__ = ('connection')
+    schema_version = 1
```

Comment:
This will cause the db to be repeatedly rebuilt if the use has venvs with different python versions (fontList.json currently suffers from the same problem).
Perhaps better would be to use something like ~/.cache/texsupport.$version.db.  This way you also don't need to handle db migration.

Reply:
That seems like a good idea.